src/render/pso_data.rs

```diff
@@ -0,0 +1,163 @@
+use color;
+use gfx::handle as h;
+use material::MaterialType;
+use render::{BackendResources, PbrParams};
+use std::mem;
+use texture::Texture;
+
+bitflags! {
+    struct PbrFlags: i32 {
+        const BASE_COLOR_MAP         = 1 << 0;
+        const NORMAL_MAP             = 1 << 1;
+        const METALLIC_ROUGHNESS_MAP = 1 << 2;
+        const EMISSIVE_MAP           = 1 << 3;
+        const OCCLUSION_MAP          = 1 << 4;
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct PbrMaps {
+    base_color: Option<Texture<[f32; 4]>>,
+    normal: Option<Texture<[f32; 4]>>,
+    emissive: Option<Texture<[f32; 4]>>,
+    metallic_roughness: Option<Texture<[f32; 4]>>,
+    occlusion: Option<Texture<[f32; 4]>>,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct PbrMapParams {
+    pub(crate) base_color: (
+        h::ShaderResourceView<BackendResources, [f32; 4]>,
```

Comment:
let's have a type for shader view + sampler, given that we need it many times here